extensions/wikia/Privacy/RemoveUserDataController.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+use Wikia\Logger\Loggable;
+
+class RemoveUserDataController extends WikiaController {
+	use Loggable;
+
+	const ACCEPTED = 202;
+	const METHOD_NOT_ALLOWED = 405;
+
+	public function allowsExternalRequests() {
+		return false;
+	}
+
+	public function removeUserData() {
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+
+		if ( !$this->request->wasPosted() ) {
+			$this->response->setCode( self::METHOD_NOT_ALLOWED );
+			return;
+		}
+
+		$userId = $this->getVal( 'userId' );
+		if ( empty( $userId ) ) {
+			$this->response->setCode( WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+			return;
+		}
+
+		$this->info( "Right to be forgotten request for user $userId", ['userId' => $userId] );
+
+		$dataRemover = new UserDataRemover();
+		$dataRemover->removeGlobalData( $userId );
+
+		$this->info( "Global data removed for user $userId", ['userId' => $userId] );
+
+		$userWikis = $this->getUserWikis( $userId );
+		$username = User::whoIs( $userId );
+		$wikiCount = count( $userWikis );
+
+		$removeWikiDataTask = new RemoveUserDataOnWikiTask();
+		$removeWikiDataTask->call( 'removeAllData', $userId,  $username);
+		$removeWikiDataTask->wikiId( $userWikis )->queue();
+
+		$this->info( "Wiki data removal queued for user $userId", ['userId' => $userId] );
+
+		$this->response->setCode( self::ACCEPTED );
+		$this->response->setValues( [
+			'userId' => $userId,
+			'username' => $username,
+			'numberOfWikis' => $wikiCount
+		] );
+	}
+
+	/**
+	 * Helper function to assess the scope of removal.
+	 */
+	public function getRemoveDetails() {
+		$this->response->setFormat( WikiaResponse::FORMAT_JSON );
+
+		$userId = $this->getVal( 'userId' );
+		if ( empty( $userId ) ) {
+			$this->response->setCode( WikiaResponse::RESPONSE_CODE_BAD_REQUEST );
+			return;
+		}
+
+		$this->response->setCode( self::ACCEPTED );
```

Comment:
HTTP 200 fits better here :)

Reply:
I disagree. The queued tasks may still be in progress.

Reply:
But this request just gets some stats.

Reply:
Ah right, my bad